### PR TITLE
Fixed `repr` for `SecondaryAxis`

### PR DESCRIPTION
**Pull request by oscargus**
and increase coverage for axes/_base.py

## PR Summary

Main purpuse was to increase test coverage, but when I played around, I found the following:

```
>>> ax.secondary_xaxis('bottom')
Traceback (most recent call last):

  File ~\miniconda3\lib\site-packages\IPython\core\formatters.py:707 in __call__
    printer.pretty(obj)

  File ~\miniconda3\lib\site-packages\IPython\lib\pretty.py:410 in pretty
    return _repr_pprint(obj, self, cycle)

  File ~\miniconda3\lib\site-packages\IPython\lib\pretty.py:778 in _repr_pprint
    output = repr(obj)

  File ~\matplotlib\lib\matplotlib\axes\_base.py:711 in __repr__
    title = self.get_title(loc=k)

AttributeError: 'SecondaryAxis' object has no attribute 'get_title'
```

so I fixed that as well.

## PR Checklist

**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [N/A] New features are documented, with examples if plot related.
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [X] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/axes/_base.py [hunk 1]
```diff
@@ -708,7 +708,10 @@ def __repr__(self):
             fields += [f"label={self.get_label()!r}"]
         titles = []
         for k in ["left", "center", "right"]:
-            title = self.get_title(loc=k)
+            if hasattr(self, 'get_title'):
+                title = self.get_title(loc=k)
+            else:
+                title = ""
             if title:
                 titles.append(f"{k!r}:{title!r}")
```
Comment (timhoffm) [inline lib/matplotlib/axes/_base.py:716]: Slightly simpler:
```suggestion
            if hasattr(self, 'get_title'):
                title = self.get_title(loc=k)
                if title:
                    titles.append(f"{k!r}:{title!r}")
```

File: lib/matplotlib/tests/test_axes.py [hunk 2]
```diff
@@ -489,6 +489,20 @@ def test_autoscale_tight():
     assert_allclose(ax.get_xlim(), (-0.15, 3.15))
     assert_allclose(ax.get_ylim(), (1.0, 4.0))
 
+    # Check that autoscale is on
+    assert ax.get_autoscalex_on() is True
+    assert ax.get_autoscaley_on() is True
+    assert ax.get_autoscale_on() is True
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_axes.py:495]:
```suggestion
    assert ax.get_autoscalex_on()
    assert ax.get_autoscaley_on()
    assert ax.get_autoscale_on()
```

File: lib/matplotlib/tests/test_axes.py [hunk 3]
```diff
@@ -489,6 +489,20 @@ def test_autoscale_tight():
...
+    # Set enable to None
+    ax.autoscale(enable=None)
+    # Same limits
+    assert_allclose(ax.get_xlim(), (-0.15, 3.15))
+    assert_allclose(ax.get_ylim(), (1.0, 4.0))
+    # autoscale still on
+    assert ax.get_autoscalex_on() is True
+    assert ax.get_autoscaley_on() is True
+    assert ax.get_autoscale_on() is True
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_axes.py:504]:
```suggestion
    assert ax.get_autoscalex_on()
    assert ax.get_autoscaley_on()
    assert ax.get_autoscale_on()
```

File: lib/matplotlib/axes/_base.py (hunk 1 above)
Comment (oscargus) [reply lib/matplotlib/axes/_base.py:716]: Yeah, makes sense!

File: lib/matplotlib/tests/test_axes.py (hunk 2 above)
Comment (oscargus) [reply lib/matplotlib/tests/test_axes.py:495]: OK! Wasn't sure what the preferred way was.

**Review by timhoffm — APPROVED** [review]

**Review by greglucas — APPROVED** [review]

**Comment by QuLogic** [comment]
@meeseeksdev backport to v3.5.x

**Comment by QuLogic** [comment]
OK, I won't bother then.

**Comment by oscargus** [comment]
Should I backport manually? No problems. Most likely it is a conflict in the tests, so should be easy to fix.

**Comment by tacaswell** [comment]
@oscargus If you want to do the backport we will definitely merge it.

**Pull request merged**